packages/protocol/contracts/governance/DoubleSigningSlasher.sol

```diff
@@ -0,0 +1,163 @@
+pragma solidity ^0.5.3;
+
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+import "../common/Initializable.sol";
+import "../common/UsingRegistry.sol";
+import "../common/UsingPrecompiles.sol";
+
+contract DoubleSigningSlasher is Ownable, Initializable, UsingRegistry, UsingPrecompiles {
+  struct SlashingIncentives {
+    // Value of LockedGold to slash from the account.
+    uint256 penalty;
+    // Value of LockedGold to send to the observer.
+    uint256 reward;
+  }
+
+  SlashingIncentives public slashingIncentives;
+  mapping(bytes32 => bool) isSlashed;
+
+  event SlashingIncentivesSet(uint256 penalty, uint256 reward);
+
+  function initialize(address registryAddress, uint256 _penalty, uint256 _reward)
+    external
+    initializer
+  {
+    _transferOwnership(msg.sender);
+    setRegistry(registryAddress);
+    _setSlashingIncentives(_penalty, _reward);
+  }
+
+  // Require that `reward` is less than `penalty`.
+  function setSlashingIncentives(uint256 penalty, uint256 reward) external onlyOwner {
+    _setSlashingIncentives(penalty, reward);
+  }
+
+  // Require that `reward` is less than `penalty`.
+  function _setSlashingIncentives(uint256 penalty, uint256 reward) internal {
```

Comment:
is separating the `external modifier` functions from the `internal` logic defined in a style guide somewhere? I started seeing this recently but not sure what I prefer

Reply:
We've been using it in cases where the external function is `onlyVm()`, so that we can have a testing contract that exposes the function without `onlyVm()`. Haven't read the code so not yet sure if that applies here